packages/admin/cms-admin/src/pages/pageTree/PageTree.tsx

```diff
@@ -174,6 +204,15 @@ const PageTree: React.ForwardRefRenderFunction<PageTreeRefApi, PageTreeProps> =
 
     const propsForVirtualList = useDndWindowScroll();
 
+    React.useEffect(() => {
+        if (newPageIds.current.length > 0) {
+            const index = pages.findIndex((page) => newPageIds.current.includes(page.id));
+            refList.current?.scrollToItem(index, "smart");
```

Comment:
Why don't you implement the scrollToItem call in the query watcher? This would remove the need for the `newPageIds` ref and the `useEffect`

Reply:
I need the useRef() anyway to check if a table row should slide in 

```tsx
    slideIn={newPageIds.current.includes(page.id)}
    // https://github.com/vivid-planet/comet/pull/652/files#diff-c15ff74463dbeff193f9735495def10cb7974b702b5515cadb3ace52806aa7c2R246
```

Reply:
Also, this wouldn't work correctly because the at the time the watcher callback is executed, the `pages` prop has not yet been updated. Therefore, the new element doesn't exist in `pages` yet => I can't figure out the index